code/graphics/debug_sphere.cpp

```diff
@@ -0,0 +1,26 @@
+#include "debug_sphere.h"
+#include "globalincs/vmallocator.h"
+#include "render/3d.h"
+
+namespace debug_spheres{
+
+SCP_vector<debug_sphere> Spheres;
+
+void add(vec3d pos, float rad){
+	debug_sphere s;
+	s.pos = pos;
+	s.radius = rad;
+	Spheres.push_back(s);
+}
+
+void clear(){
+	Spheres.clear();
+}
+void render(){
+	for(auto s : Spheres){
+		g3_draw_sphere_ez(&(s.pos),s.radius);
+	}
+	clear();
```

Comment:
It's an interesting idea to just clear the whole vector every time.  Maybe, if expanded in the future, you could have the option of having a sphere persist.

Reply:
Clearing every frame mimics how lights are implemented. There's definitely room here for some more logic in the future, yeah.